src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/utils/validation.ts

```diff
@@ -454,6 +455,34 @@ export function validateFormComponentsForLayout(
           validations[component.id] = componentValidations;
         }
       }
+      if (component.type === 'FileUploadWithTag') {
+        const isValid = attachmentsValid(attachments, component);
+        const missingTagAttachments = attachments[component.id]
+          .filter((attachment) => attachmentIsMissingTag(attachment))
+          .map((attachment) => attachment.id);
+        if (!isValid || missingTagAttachments.length > 0) {
+          validations[component.id] = {};
+          const componentValidations: IComponentValidations = {
+            [fieldKey]: {
+              errors: [],
+              warnings: [],
+            },
+          };
+          if (!isValid) {
+            componentValidations[fieldKey].errors.push(
+              `${getLanguageFromKey('form_filler.file_uploader_validation_error_file_number_1', language)} ${component.minNumberOfAttachments} ${getLanguageFromKey('form_filler.file_uploader_validation_error_file_number_2', language)}`,
+            );
+          }
+          if (missingTagAttachments.length > 0) {
+            missingTagAttachments.forEach((missingId) => {
+              componentValidations[fieldKey].errors.push( // Add attachment id to start of message and seperate using ASCII Universal Seperator.
```

Comment:
Why are we separating with ASCII Universal Separator? Is it a pattern we are using elsewhere, or some standard?

I see its used like this in other places in this PR as well. One thing I think would be good would be to create a util const of `String.fromCharCode(31)`, so we dont have to explain what it is with a comment every time its used. 

Could f.ex put something like this in a util: 

```jsx
export const AsciiUniversalSeparator = String.fromCharCode(31);
```

Reply:
It is a standard in ASCII at least. It may be a little hacky but needed a delimiter for the id and message. Unless we want a rewrite of how error messages are handled this seemed like a worthwhile workaround.
I've implemented it in a util